src/CMakeLists.txt

```diff
@@ -278,10 +278,12 @@ list(SORT SRC_H)
 
 # On Windows use specific platform sources
 if(WIN32 AND NOT CYGWIN)
-	if(MSVC)
-		SET(WIN_RC "win32/git2.rc")
+	if(NOT MSVC)
+		ADD_DEFINITIONS(-DGCC_WINDRES)
```

Comment:
I've used the same approach as with zlib which explicitly adds this define specifically for the resource script: https://github.com/madler/zlib/blob/v1.2.11/CMakeLists.txt#L178

What I like in this approach is that this change is then very contained, not affecting anything else in the codebase.

If you still think that it should be replaced by any built-in macro, I will happily change it :)

Reply:
Yeah, I would like to use an existing type like `_MSC_VER` (though I'm not tied to it necessarily _being_ `_MSC_VER`, that was just an example).  The reason being that there are people who are building libgit2 by including the code directly in their project, without using cmake at all.  It's unlikely that they're going to start using gcc to do that build where they weren't before, but it would be nice to have that scenario covered for them so that there's one fewer hurdle.  🙏 

Otherwise, we should probably include this in `feature.h` so that they can configure it directly, but I think that re-using an existing definition is the better solution.

Reply:
Sorry for the delayed response; I've checked the `_MSC_VER` macro and it's not defined for the .rc file. I think it is because the .rc file is compiled by the Resource Compiler (rc.exe) and not MSVC compiler.

There is however a way to pass a define to the `windres.exe` GCC resource compiler. Regarding that currently the .rc file cannot be compiled by windres at all, I think that people who are building libgit2 without CMake are not including the .rc file at all. Maybe we could just add a note to the readme that the .rc file can now be compiled by windres as long as `GCC_WINDRES` macro is defined?

Reply:
`__GNUC__` seems to work though 🤔